src/evaluatorHelpers.ts

```diff
@@ -251,7 +251,18 @@ export async function renderPrompt(
     // Recursively walk the JSON structure. If we find a string, render it with nunjucks.
     return JSON.stringify(renderVarsInObject(parsed, vars), null, 2);
   } catch {
-    return renderVarsInObject(nunjucks.renderString(basePrompt, vars), vars);
+    // Vars values can be template strings, so we need to render them first:
+    const renderedVars = Object.fromEntries(
+      Object.entries(vars).map(([key, value]) => [
+        key,
+        typeof value === 'string' ? nunjucks.renderString(value, vars) : value,
+      ]),
+    );
```

Comment:
Not sure if I follow- we're rendering the vars, and then rendering the base prompt itself? What case is that handling, and is that captured in a test?

Reply:
```
{
  basePrompt: 'Write {{ headline }}',
  vars: { headline: 'Articles about {{ env.SENTINAL }}' },
  renderedVars: { headline: 'Articles about foobar' }
}
```